contrib/platform/src/com/sun/jna/platform/linux/LibC.java

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2017 Daniel Widdis, All Rights Reserved
+ * 
+ * The contents of this file is dual-licensed under 2 
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and 
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ * 
+ * You can freely decide which license you want to apply to 
+ * the project.
+ * 
+ * You may obtain a copy of the LGPL License at:
+ * 
+ * http://www.gnu.org/licenses/licenses.html
+ * 
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ * 
+ * You may obtain a copy of the Apache License at:
+ * 
+ * http://www.apache.org/licenses/
+ * 
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.linux;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import com.sun.jna.NativeLong;
+import com.sun.jna.Structure;
+import com.sun.jna.Structure.FieldOrder;
+import com.sun.jna.platform.unix.LibCAPI;
+
+/**
+ * <I>libc</I> API
+ * 
+ * @author Daniel Widdis
+ */
+public interface LibC extends LibCAPI, Library {
+    String NAME = "c";
+    LibC INSTANCE = Native.load(NAME, LibC.class);
+
+    @FieldOrder({ "uptime", "loads", "totalram", "freeram", "sharedram", "bufferram", "totalswap", "freeswap", "procs",
+            "totalhigh", "freehigh", "mem_unit", "_f" })
+    class Sysinfo extends Structure {
+        public NativeLong uptime; // Seconds since boot
+        // 1, 5, and 15 minute load averages
+        public NativeLong[] loads = new NativeLong[3];
+        public NativeLong totalram; // Total usable main memory size
+        public NativeLong freeram; // Available memory size
+        public NativeLong sharedram; // Amount of shared memory
+        public NativeLong bufferram; // Memory used by buffers
+        public NativeLong totalswap; // Total swap space size
+        public NativeLong freeswap; // swap space still available
+        public short procs; // Number of current processes
+        public NativeLong totalhigh; // Total high memory size
+        public NativeLong freehigh; // Available high memory size
+        public int mem_unit; // Memory unit size in bytes
+        // Padding is based on the size of NativeLong. When the size is 4 we
+        // need 8 bytes of padding. When this size is 8, zero padding is needed
+        // but we are not allowed to declare an array of size zero. Here we
+        // declare the array but will override calculateSize to ensure the
+        // correct C side allocation occurs. These bytes are never read.
+        public byte[] _f = new byte[8]; // padding to 64 bytes
```

Comment:
This is not correct. The target size on 32 bit is 64bytes. But is now 72. With the _f member the structure is 8 bytes to large, as the padding happens via the calculateSize and the member.

I would remove the _f member. The memory needs to allocated and be present, but it is not needed to be mapped.

Reply:
Oops. I just copied your block of code without checking it.  The conditional should be based on NativeLong.SIZE rather than CALCULATE_SIZE.   But also the array calculation, starting in the superclass, would stay in the superclass.  We would have to override `toArray()` for that approach to work.

Leaving `_f` off would leave the size 8 bytes too short on 32-bit systems without the `toArray()` override.

Perhaps the best answer is two separate structures, a `SysInfo` and `SysInfo64` and have the caller do a size check to decide which one to use...